openmls/src/group/core_group/test_external_init.rs

```diff
@@ -167,6 +169,107 @@ fn test_external_init(ciphersuite: &'static Ciphersuite, backend: &impl OpenMlsC
         group_charly.treesync().export_nodes(),
         group_bob.treesync().export_nodes()
     );
+
+    // Check if charly can create valid commits
+    let proposal_store = ProposalStore::default();
+    let params = CreateCommitParams::builder()
+        .framing_parameters(framing_parameters)
+        .credential_bundle(&charly_credential_bundle)
+        .proposal_store(&proposal_store)
+        .build();
+    let create_commit_result = group_charly
+        .create_commit(params, backend)
+        .expect("Error creating commit");
+
+    let staged_commit = group_alice
+        .stage_commit(&create_commit_result.commit, &proposal_store, &[], backend)
+        .expect("error staging commit");
+    group_alice
+        .merge_commit(staged_commit)
+        .expect("error merging commit");
+
+    group_charly
+        .merge_commit(create_commit_result.staged_commit)
+        .expect("error merging commit");
+
+    // Now we assume that Bob somehow lost his group state and wants to add
+    // themselves back through an external commit.
+
+    // Have Alice export everything that Bob needs.
+    let pgs_encoded: Vec<u8> = group_alice
+        .export_public_group_state(backend, &alice_credential_bundle)
+        .expect("Error exporting PGS")
+        .tls_serialize_detached()
+        .expect("Error serializing PGS");
+    let verifiable_public_group_state =
+        VerifiablePublicGroupState::tls_deserialize(&mut pgs_encoded.as_slice())
+            .expect("Error deserializing PGS");
+    let nodes_option = group_alice.treesync().export_nodes();
+
+    let proposal_store = ProposalStore::new();
+    let params = CreateCommitParams::builder()
+        .framing_parameters(framing_parameters)
+        .credential_bundle(&bob_credential_bundle)
+        .proposal_store(&proposal_store)
+        .build();
+    println!("\nBob joins externally.");
+    let (mut new_group_bob, create_commit_result) = CoreGroup::join_by_external_commit(
+        backend,
+        params,
+        Some(&nodes_option),
```

Comment:
Not part of this PR, but I'm wondering how you could join a group without providing any nodes.

Reply:
You can't, same as with a `Welcome`, I think. This is only an `Option`, because the tree could be in a RatchetTreeExtension instead.